serve/mlc_serve/engine/sync_engine.py

```diff
@@ -157,11 +159,15 @@ def step(self) -> InferenceStepResult:
                         error=res.error,
                     )
                 )
-                continue
+            else:
+                valid_results.append(res)
 
+        for res in valid_results:
```

Comment:
staged engine has the same loop, need to handle in it as well

Reply:
No. The sync engine is accessing `self.current_batch[request_id]`, but for a failed request this element is deleted before this access is made. Check the following two lines.

https://github.com/octoml/mlc-llm/blob/batch-serving/serve/mlc_serve/engine/sync_engine.py#L151
https://github.com/octoml/mlc-llm/blob/batch-serving/serve/mlc_serve/engine/sync_engine.py#L177

The staging engine doesn't have this problem.